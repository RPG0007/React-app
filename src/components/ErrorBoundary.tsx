import { Component, ErrorInfo } from 'react';

interface IProps {
  children?: React.ReactNode;
}

interface IState {
  hasError: boolean;
  error: null | Error;
  errorInfo: null | ErrorInfo;
}

export default class ErrorBoundary extends Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    this.state = {
      hasError: false,
      error: null,
      errorInfo: null,
    };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    this.setState({
      hasError: true,
      error,
      errorInfo,
    });
  }

  render() {
    if (this.state.hasError) {
      return (
        <h2 className="title">
          An unexpected error has occurred, please reload the page.
        </h2>
      );
    }

    return this.props.children;
  }
}
