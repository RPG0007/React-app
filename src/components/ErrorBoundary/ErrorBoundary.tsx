import { Component, ErrorInfo } from 'react';
import {
  IErrorBoundaryProps,
  IErrorBoundaryState,
} from '../../types/interfaces';

export default class ErrorBoundary extends Component<
  IErrorBoundaryProps,
  IErrorBoundaryState
> {
  constructor(props: IErrorBoundaryProps) {
    super(props);
    this.state = {
      hasError: false,
      error: null,
      errorInfo: null,
    };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    this.setState({
      hasError: true,
      error,
      errorInfo,
    });
  }

  render() {
    if (this.state.hasError) {
      return (
        <h2 className="title">
          An unexpected error has occurred, please reload the page.
        </h2>
      );
    }

    return this.props.children;
  }
}
