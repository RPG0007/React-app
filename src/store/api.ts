import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';
import { BASE_URL } from '../constants/constants';
import { ICardDescription, IResponseApi } from '../types/interfaces';

export const rickAndMortyApi = createApi({
  reducerPath: 'rickAndMortyApi',
  baseQuery: fetchBaseQuery({ baseUrl: BASE_URL }),
  endpoints: (builder) => ({
    getResultNewSearch: builder.query<IResponseApi, string>({
      query: (queryString) => `${queryString}`,
     

  }),
    getCardDesctiption: builder.query<ICardDescription, string>({
      query: (queryString) => `${queryString}`,
    }),
  }),
});

export const { useGetResultNewSearchQuery, useLazyGetCardDesctiptionQuery } =
  rickAndMortyApi;
  
  export const { getResultNewSearch, getCardDesctiption } = rickAndMortyApi.endpoints;
