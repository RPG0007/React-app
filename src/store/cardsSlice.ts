import { createSlice } from '@reduxjs/toolkit';
import type { PayloadAction } from '@reduxjs/toolkit';
import { Cards } from '../types/interfaces';

interface ICardsState {
  cards: Cards;
  isCardsLoading: boolean;
  searchString: string;
  prevSearchString: string;
  isNewSearchCalled: number;
}

const initialState: ICardsState = {
  cards: [],
  isCardsLoading: true,
  searchString: '',
  prevSearchString: '',
  isNewSearchCalled: 0,
};

export const cardsSlice = createSlice({
  name: 'cards',
  initialState,
  reducers: {
    changeCurrentPageCards: (state, action: PayloadAction<Cards>) => {
      state.cards.length = 0;
      action.payload.map((el) => {
        state.cards.push(el);
      });
    },
    changeIsCardsLoading: (state, action: PayloadAction<boolean>) => {
      state.isCardsLoading = action.payload;
    },
    changeSearchString: (state, action: PayloadAction<string>) => {
      state.searchString = action.payload;
    },
    changePrevSearchString: (state, action: PayloadAction<string>) => {
      state.prevSearchString = action.payload;
    },
    changeIsNewSearchCalled: (state, action: PayloadAction<number>) => {
      state.isNewSearchCalled = action.payload;
    },
  },
});

export const {
  changeCurrentPageCards,
  changeIsCardsLoading,
  changeSearchString,
  changeIsNewSearchCalled,
  changePrevSearchString,
} = cardsSlice.actions;

// Other code such as selectors can use the imported `RootState` type
//export const selectCount = (state: RootState) => state.counter.value

export default cardsSlice.reducer;
