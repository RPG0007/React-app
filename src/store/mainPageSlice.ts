import { createSlice } from '@reduxjs/toolkit';
import type { PayloadAction } from '@reduxjs/toolkit';
import { Cards, ICardDescription } from '../types/interfaces';

interface IMainPageState {
  cards: Cards;
  cardDescription: ICardDescription | undefined;
  isCardsLoading: boolean;
  isModalActive: boolean;
  isModalLoading: boolean;
  searchString: string;
  allPage: number;
  numPerpage: number;
}

const initialState: IMainPageState = {
  cards: [],
  cardDescription: undefined,
  isCardsLoading: true,
  isModalActive: false,
  isModalLoading: false,
  searchString: '',
  allPage: 1,
  numPerpage: 20,
};

export const mainPageSlice = createSlice({
  name: 'cards',
  initialState,
  reducers: {
    changeCurrentCards: (state, action: PayloadAction<Cards>) => {
      state.cards = action.payload;
    },
    changeCardDescription: (
      state,
      action: PayloadAction<ICardDescription | undefined>
    ) => {
      state.cardDescription = action.payload;
    },
    changeIsCardsLoading: (state, action: PayloadAction<boolean>) => {
      state.isCardsLoading = action.payload;
    },
    changeIsModalLoading: (state, action: PayloadAction<boolean>) => {
      state.isModalLoading = action.payload;
    },
    changeIsModalActive: (state, action: PayloadAction<boolean>) => {
      state.isModalActive = action.payload;
    },
    changeSearchString: (state, action: PayloadAction<string>) => {
      state.searchString = action.payload;
    },
    changeAllPage: (state, action: PayloadAction<number>) => {
      state.allPage = action.payload;
    },
    changenumPerpage: (state, action: PayloadAction<number>) => {
      state.numPerpage = action.payload;
    },
  },
});

export const {
  changeCurrentCards,
  changeCardDescription,
  changeIsCardsLoading,
  changeIsModalLoading,
  changeIsModalActive,
  changeSearchString,
  changeAllPage,
  changenumPerpage,
} = mainPageSlice.actions;

export default mainPageSlice.reducer;
