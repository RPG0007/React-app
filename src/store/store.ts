import { configureStore } from '@reduxjs/toolkit';
import mainPageSlice from './mainPageSlice';
import { rickAndMortyApi } from './api';
import { MakeStore, createWrapper } from 'next-redux-wrapper';

export const store = configureStore({
  reducer: {
    mainPage: mainPageSlice,
    [rickAndMortyApi.reducerPath]: rickAndMortyApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(rickAndMortyApi.middleware),
});

const makeStore: MakeStore<AppStore> = () => store;
export type RootState = ReturnType<typeof store.getState>;
export type AppStore = typeof store;
export type AppDispatch = AppStore['dispatch'];
export const wrapper = createWrapper<AppStore>(makeStore, { debug: true });
