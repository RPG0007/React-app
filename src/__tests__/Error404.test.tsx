import React from 'react';
import { describe, expect, it, vi } from 'vitest';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import ErrorPage from '../pages/404';
import Home, { getServerSideProps } from '@/pages';
import { CardRequest } from './_fakeData';
import ErrorBoundary from '@/components/ErrorBoundary/ErrorBoundary';
import { Provider } from 'react-redux';
import { store } from '@/store/store';
import { GetServerSidePropsContext } from 'next';
import { ParsedUrlQuery } from 'querystring';

vi.mock('next/router', () => vi.importActual('next-router-mock'));

describe('Tests for the 404 Page component', () => {
  it('404 page is displayed when navigating to an invalid route.', async () => {
    render(<ErrorPage />);
    expect(screen.getByTestId('404-error-page')).toBeInTheDocument();
   
  });

 

  it('renders snapshot of app component', () => {
    const tree = render(
      <ErrorBoundary>
      <Provider store={store}> 
    <Home datar={CardRequest} />
    </Provider>
    </ErrorBoundary>)
    console.log(tree)
    expect(tree).toMatchSnapshot()
  })
  it('prepares pageProps for component', async () => {
    const context2 = {
        query: { page:'1',name:"",id: "fjdks" } as ParsedUrlQuery
    };
     const value2 =  getServerSideProps(context2 as GetServerSidePropsContext);
     expect (value2).toBeTruthy();

  })
});