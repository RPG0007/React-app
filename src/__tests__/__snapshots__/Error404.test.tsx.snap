// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests for the 404 Page component > renders snapshot of app component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="_main_b140de"
      >
        <div
          class="_description_b140de"
        >
          <div
            class="_searchSection_a9d8a5"
          >
            <input
              class="_searchInput_a9d8a5"
              data-testid="input-search"
              placeholder="search by name"
              type="search"
              value=""
            />
            <button
              class="_searchButton_a9d8a5"
              data-testid="input-btn"
            >
              <img
                alt="search icon"
                class="_searchIcon_a9d8a5"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="40"
                src="/_next/image?url=%2Fsrc%2Fassets%2Fsearch.png&w=96&q=75"
                srcset="/_next/image?url=%2Fsrc%2Fassets%2Fsearch.png&w=48&q=75 1x, /_next/image?url=%2Fsrc%2Fassets%2Fsearch.png&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
            </button>
          </div>
          <button
            class="_errorButton_518232"
          >
            Error button
          </button>
        </div>
        <div
          class="cardsWrapper"
        >
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
            />
            <h2>
              Rick Sanchez
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
            />
            <h2>
              Morty Smith
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
            />
            <h2>
              Rick Sanchez
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
            />
            <h2>
              Morty Smith
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
            />
            <h2>
              Rick Sanchez
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
            />
            <h2>
              Morty Smith
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
            />
            <h2>
              Rick Sanchez
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
            />
            <h2>
              Morty Smith
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
            />
            <h2>
              Rick Sanchez
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
            />
            <h2>
              Morty Smith
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
            />
            <h2>
              Rick Sanchez
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
            />
            <h2>
              Morty Smith
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
            />
            <h2>
              Rick Sanchez
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
            />
            <h2>
              Morty Smith
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
            />
            <h2>
              Rick Sanchez
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
            />
            <h2>
              Morty Smith
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
            />
            <h2>
              Rick Sanchez
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
            />
            <h2>
              Morty Smith
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
            />
            <h2>
              Rick Sanchez
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
          <div
            class="_card_327d43"
            data-testid="card"
          >
            <img
              alt="image character"
              class="_cardImg_327d43"
              src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
            />
            <h2>
              Morty Smith
            </h2>
            <h4>
              species: Human
            </h4>
            <h4>
              gender: Male
            </h4>
            <h4>
              status: Alive
            </h4>
          </div>
        </div>
        <div
          class="_paginationSection_4a2aa2"
          data-testid="pagination"
        >
          <button
            class="_paginationButton_4a2aa2 disabledButton"
            data-testid="button-prev-page"
            disabled=""
          >
            &lt;
          </button>
          <h4
            class="paginationPage"
          >
            1
             / 
            1
          </h4>
          <button
            class="_paginationButton_4a2aa2 disabledButton"
            data-testid="button-next-page"
            disabled=""
          >
            &gt;
          </button>
          <input
            class="_pageInput_4a2aa2"
            max="20"
            placeholder="max 20 per page"
            type="text"
          />
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="_main_b140de"
    >
      <div
        class="_description_b140de"
      >
        <div
          class="_searchSection_a9d8a5"
        >
          <input
            class="_searchInput_a9d8a5"
            data-testid="input-search"
            placeholder="search by name"
            type="search"
            value=""
          />
          <button
            class="_searchButton_a9d8a5"
            data-testid="input-btn"
          >
            <img
              alt="search icon"
              class="_searchIcon_a9d8a5"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="40"
              src="/_next/image?url=%2Fsrc%2Fassets%2Fsearch.png&w=96&q=75"
              srcset="/_next/image?url=%2Fsrc%2Fassets%2Fsearch.png&w=48&q=75 1x, /_next/image?url=%2Fsrc%2Fassets%2Fsearch.png&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </button>
        </div>
        <button
          class="_errorButton_518232"
        >
          Error button
        </button>
      </div>
      <div
        class="cardsWrapper"
      >
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          />
          <h2>
            Rick Sanchez
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
          />
          <h2>
            Morty Smith
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          />
          <h2>
            Rick Sanchez
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
          />
          <h2>
            Morty Smith
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          />
          <h2>
            Rick Sanchez
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
          />
          <h2>
            Morty Smith
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          />
          <h2>
            Rick Sanchez
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
          />
          <h2>
            Morty Smith
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          />
          <h2>
            Rick Sanchez
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
          />
          <h2>
            Morty Smith
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          />
          <h2>
            Rick Sanchez
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
          />
          <h2>
            Morty Smith
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          />
          <h2>
            Rick Sanchez
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
          />
          <h2>
            Morty Smith
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          />
          <h2>
            Rick Sanchez
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
          />
          <h2>
            Morty Smith
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          />
          <h2>
            Rick Sanchez
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
          />
          <h2>
            Morty Smith
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          />
          <h2>
            Rick Sanchez
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
        <div
          class="_card_327d43"
          data-testid="card"
        >
          <img
            alt="image character"
            class="_cardImg_327d43"
            src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
          />
          <h2>
            Morty Smith
          </h2>
          <h4>
            species: Human
          </h4>
          <h4>
            gender: Male
          </h4>
          <h4>
            status: Alive
          </h4>
        </div>
      </div>
      <div
        class="_paginationSection_4a2aa2"
        data-testid="pagination"
      >
        <button
          class="_paginationButton_4a2aa2 disabledButton"
          data-testid="button-prev-page"
          disabled=""
        >
          &lt;
        </button>
        <h4
          class="paginationPage"
        >
          1
           / 
          1
        </h4>
        <button
          class="_paginationButton_4a2aa2 disabledButton"
          data-testid="button-next-page"
          disabled=""
        >
          &gt;
        </button>
        <input
          class="_pageInput_4a2aa2"
          max="20"
          placeholder="max 20 per page"
          type="text"
        />
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
